programs/amm/src/states/pool.rs

```diff
@@ -523,9 +523,8 @@ impl PoolState {
             }
             last_tick_array_start_index = start_index;
 
-            let Some(tickarray_bitmap_extension) = tickarray_bitmap_extension else {
```

Comment:
getting `if else` not supported by older version of rust, seems to be caused by solana 1.14.18. upgrading to 1.16.15 is causing a lot of problems.